src/com/namelessmc/java_api/NamelessAPI.java

```diff
@@ -97,50 +91,44 @@ static String getApiKey(final String url) {
 	public void checkWebAPIConnection() throws NamelessException {
 		final JsonObject response = this.requests.get(Action.INFO);
 		if (!response.has("nameless_version")) {
-			throw new NamelessException("Invalid respose: " + response.getAsString());
+			throw new NamelessException("Invalid response: " + response.getAsString());
 		}
 	}
 
 	/**
 	 * Get all announcements
+	 *
 	 * @return list of current announcements
 	 * @throws NamelessException if there is an error in the request
 	 */
 	public List<Announcement> getAnnouncements() throws NamelessException {
 		final JsonObject response = this.requests.get(Action.GET_ANNOUNCEMENTS);
 
-		final List<Announcement> announcements = new ArrayList<>();
-
-		response.getAsJsonArray().forEach((element) -> {
-			final JsonObject announcementJson = element.getAsJsonObject();
-			final String content = announcementJson.get("content").getAsString();
-			final String[] display = jsonToArray(announcementJson.get("display").getAsJsonArray());
-			final String[] permissions = jsonToArray(announcementJson.get("permissions").getAsJsonArray());
-			announcements.add(new Announcement(content, display, permissions));
-		});
-
-		return announcements;
+		return jsonArrayToList(response.getAsJsonArray());
 	}
 
 	/**
 	 * Get all announcements visible for the player with the specified uuid
-	 * @param uuid UUID of player to get visibile announcements for
+	 *
+	 * @param user player to get visibile announcements for
 	 * @return list of current announcements visible to the player
 	 * @throws NamelessException if there is an error in the request
 	 */
 	public List<Announcement> getAnnouncements(final NamelessUser user) throws NamelessException {
 		final JsonObject response = this.requests.get(Action.GET_ANNOUNCEMENTS, "id", user.getId());
 
-		final List<Announcement> announcements = new ArrayList<>();
+		return jsonArrayToList(response.get("announcements").getAsJsonArray());
+	}
 
-		response.get("announcements").getAsJsonArray().forEach((element) -> {
+	private List<Announcement> jsonArrayToList(JsonArray array) {
```

Comment:
could you call this function `jsonArrayToAnnouncementsList` to show that it is specifically for announcements